File: BookLibraryProject/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookLibraryProject.Models
{
    public class Book
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int BookId { get; set; }

        [Required]
        [StringLength(100)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string LastName { get; set; } = string.Empty;

        public int TotalCopies { get; set; } = 0;

        public int CopiesInUse { get; set; } = 0;

        [StringLength(50)]
        public string? Type { get; set; }

        [StringLength(80)]
        public string? ISBN { get; set; }

        [StringLength(50)]
        public string? Category { get; set; }
               
        [NotMapped]
        public string FullAuthorName => $"{FirstName} {LastName}";
                
        [NotMapped]
        public int AvailableCopies => TotalCopies - CopiesInUse;
    }
}

File: WebApp/book-library-app/src/types/book.ts
export interface Book {
  bookId: number;
  title: string;
  firstName: string;
  lastName: string;
  fullAuthorName: string;
  totalCopies: number;
  copiesInUse: number;
  availableCopies: number;
  type?: string;
  isbn?: string;
  category?: string;
}

export interface CreateBookRequest {
  title: string;
  firstName: string;
  lastName: string;
  totalCopies: number;
  copiesInUse?: number;
  type?: string;
  isbn?: string;
  category?: string;
}

export interface UpdateBookRequest extends CreateBookRequest {
  bookId: number;
}